pkg/daemon/gpspipe.go

```diff
@@ -0,0 +1,104 @@
+package daemon
+
+import (
+	"context"
+	"fmt"
+	"github.com/golang/glog"
+	"os/exec"
+	"strings"
+	"sync"
+	"syscall"
+)
+
+const (
+	GPSPIPE_PROCESSNAME = "gpspipe"
+	GPSPIPE_SERIALPORT  = "/gpsd/data"
+)
+
+type gpspipe struct {
+	name       string
+	execMutex  sync.Mutex
+	cmd        *exec.Cmd
+	serialPort string
+	exitCh     chan bool
+	stopped    bool
+	cancelFn   context.CancelFunc
+	ctx        context.Context
+}
+
+func (gp *gpspipe) Name() string {
+	return gp.name
+}
+
+func (gp *gpspipe) SerialPort() string {
+	return gp.serialPort
+}
+func (gp *gpspipe) setStopped(val bool) {
+	gp.execMutex.Lock()
+	gp.stopped = val
+	gp.execMutex.Unlock()
+}
+
+func (gp *gpspipe) Stopped() bool {
+	gp.execMutex.Lock()
+	me := gp.stopped
+	gp.execMutex.Unlock()
+	return me
+}
+
+func (gp *gpspipe) cmdStop() {
+	glog.Infof("Stopping %s...", gp.name)
+	if gp.cmd == nil {
+		return
+	}
+
+	gp.setStopped(true)
+
+	if gp.cmd.Process != nil {
+		glog.Infof("Sending TERM to PID: %d", gp.cmd.Process.Pid)
+		gp.cmd.Process.Signal(syscall.SIGTERM)
+	}
+
+	<-gp.exitCh
+	glog.Infof("Process %d terminated", gp.cmd.Process.Pid)
+}
+
+// ubxtool -w 5 -v 1 -p MON-VER -P 29.20
+func (gp *gpspipe) cmdInit() {
+	if gp.name == "" {
+		gp.name = GPSPIPE_PROCESSNAME
+	}
+	gp.ctx, gp.cancelFn = context.WithCancel(context.Background())
+
+	cmdLine := fmt.Sprintf("gpspipe -d -r -l -o  %s", gp.SerialPort())
```

Comment:
Just FYI I only used from whe running it from the command line 
#gpspipe -r > /gpsd/data &

Reply:
Yep, I have to run this as a daemon when doing it form Golang, so -d  is for daemon, and -u depends on it.